### PR TITLE
fix: do not allow duplicate organization name

### PR DESCRIPTION
**Pull request by hrithikesh026**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Fixes in this PR:
1. No two organizations can have same `organization_name`.
2. Don't allow unknown fields in organization create and update request body.
3. Made organization name required field during new organization creation.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Manual
1. create an organization with the same name. (should fail with 4XX)
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org"
}'
```
Response:
```
{
    "error": {
        "type": "invalid_request",
        "message": "Organization with the given organization_name already exists",
        "code": "IR_38"
    }
}
```

2. send an invalid key-value pair in organization create and update call. (should fail with 4XX)
Create
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org",
    "unknown_key": "value"
}'
```
Update
```
curl --location --request PUT 'http://localhost:8080/v2/organization/org_DXkXLjRK6xIMBe6VWBnj' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "your_ org",
    "unknown_key": "value"
}'
```
Response:
```
{
    "error": "Json deserialize error: unknown field `unknown_key`, expected one of `organization_name`, `organization_details`, `metadata` at line 3 column 17"
}
```
3. Create a new organization without an organization name.(should fail with 4xx).
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    
    
}'
```
Response:
```
{
    "error": "Json deserialize error: missing field `organization_name` at line 4 column 1"
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

File: migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/up.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- Your SQL goes here
+ALTER TABLE organization
+ADD CONSTRAINT organization_organization_name_key UNIQUE (organization_name);
```
Comment (lsampras) [inline migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/up.sql:3]: how do we deal with existing duplicates?

Comment (Narayanbhat166) [reply migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/up.sql:3]: We need not add this constraint to the existing application. Can you move this to v2 migrations?

Comment (hrithikesh026) [reply migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/up.sql]: done

**Review by SanchithHegde — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

File: v2_migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/down.sql
Comment (Narayanbhat166) [inline v2_migrations/2024-09-16-110154_add-unique-constraint-to-organization-name/down.sql:1]: Can you move these migrations to the already existing migrations folder in v2?

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Pull request merged**